docs/epayment.v1.yml

```
@@ -301,69 +262,151 @@ components:
+          example: 'https://example.com/redirect?orderId=acme-shop-123-order123abc'
         userFlow:
           type: string
           enum:
             - PUSH_MESSAGE
             - NATIVE_REDIRECT
             - WEB_REDIRECT
+            - QR
           x-enum-varnames:
             - PUSH_MESSAGE
             - NATIVE_REDIRECT
             - WEB_REDIRECT
+            - QR
           example: NATIVE_REDIRECT
           description: |-
             The flow for bringing to user to the Vipps Wallet payment confirmation screen.
 
             If flow is `PUSH_MESSAGE` then one of `$.profile.sub` or `$.profile.phoneNumber` must be valid and is required.
-        paymentDescription:
+        expiresAt:
           type: string
-          description: 'message to user that will appear in the app, in the merchant portal and settlement files.'
-        callbackUrl:
+          example: '2017-07-21T17:32:28Z'
+          pattern: '^((?:(\d{4}-\d{2}-\d{2})(T|t)(\d{2}:\d{2}:\d{2}(?:\.\d+)?))(Z|z|([+-](?:2[0-3]|[01][0-9]):[0-5][0-9])))$'
+          nullable: true
+          description: |-
+            Payment will be expired at the given date and time. Format must adhere to RFC 3339. Value must be greater than 10 minutes in the future, and less than 24h in the future
+            If flow is QR, the QR-image will be returned as SVG+XML if qrFormat is not set.
+        qrFormat:
+          type: object
+          description: 'Optional, to set the format for the QR. Only applicable if user_flow is set to QR.'
+          properties:
+            format:
+              type: string
+              enum:
+                - TARGET_URL
+                - SVG+XML
+                - PNG
+              example: SVG+XML
+            size:
+              type: integer
+              example: 1024
+              description: 'Eks: 200. Then 200x200 px is set at dimension for Qr. Only applicable when format is set to PNG. Defaults to 1024 if not set'
+              minimum: 100
+              maximum: 2000
+          required:
+            - format
+        paymentDescription:
           type: string
-          description: 'This is an URL for Vipps to call at the merchant''s server to provide updated information about the order after the payment request. Domain name and context path should be provided by merchant as the value for this parameter. Vipps will add `/payments/{reference}` to the end of this URL.'
+          minLength: 3
+          maxLength: 100
+          description: 'The message to user that will appear in the app, in the merchant portal and settlement files.'
       required:
         - amount
         - paymentMethod
         - reference
         - returnUrl
         - userFlow
+    CreatePaymentResponse:
+      title: Create Payment Response
+      description: Create Payment Response object
+      type: object
+      x-examples:
+        UrlRedirect:
+          redirectUrl: 'https://landing.vipps.no?token=abc123'
```

Just using the JWT example from the API RFC to make it look sufficiently cryptic.
```suggestion
          redirectUrl: 'https://landing.vipps.no?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1Ni '
```